### DISCUSSION TITLE
Set text while typing, mention users and document storing in React

### DISCUSSION THREAD

**Discussion by junoriosity** [Milkdown/milkdown#773] [Q&A]
I would like to use Milkdown in my React frontend. Now I would like to do three things

- Can we make sure that while we are typing there is some grayish text added either added or removed after the cursor and upon hitting TAB we can include that into the text - just like in Gmail Smart-Compose
![Gmail-Smart-Compose](https://user-images.githubusercontent.com/5286536/196787876-70c799c5-f594-4783-bc1e-901e8783dac3.jpeg)
- Can we make some user mentioning like in Confluence via using @
- What would be the best way to store documents?

Many thanks for your support in advance.

**Comment by Saul-Mirone** [comment]
1. Yes, the plugin will be similar to the @milkdown/plugin-slash.
2. Yes, it's very similar to @milkdown/plugin-emoji
3. I think it's markdown. But you can also store it as HTML or JSON.
